src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/UpdateOrganizationUserCommand.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.Interfaces;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Business;
+using Bit.Core.Models.Data;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptions.Interface;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+
+namespace Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers;
+
+public class UpdateOrganizationUserCommand : IUpdateOrganizationUserCommand
+{
+    private readonly IEventService _eventService;
+    private readonly IOrganizationService _organizationService;
+    private readonly IOrganizationRepository _organizationRepository;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+    private readonly ICountNewSmSeatsRequiredQuery _countNewSmSeatsRequiredQuery;
+    private readonly IUpdateSecretsManagerSubscriptionCommand _updateSecretsManagerSubscriptionCommand;
+
+    public UpdateOrganizationUserCommand(
+        IEventService eventService,
+        IOrganizationService organizationService,
+        IOrganizationRepository organizationRepository,
+        IOrganizationUserRepository organizationUserRepository,
+        ICountNewSmSeatsRequiredQuery countNewSmSeatsRequiredQuery,
+        IUpdateSecretsManagerSubscriptionCommand updateSecretsManagerSubscriptionCommand)
+    {
+        _eventService = eventService;
+        _organizationService = organizationService;
+        _organizationRepository = organizationRepository;
+        _organizationUserRepository = organizationUserRepository;
+        _countNewSmSeatsRequiredQuery = countNewSmSeatsRequiredQuery;
+        _updateSecretsManagerSubscriptionCommand = updateSecretsManagerSubscriptionCommand;
+    }
+
+    public async Task UpdateUserAsync(OrganizationUser user, Guid? savingUserId,
+        IEnumerable<CollectionAccessSelection> collections, IEnumerable<Guid> groups)
+    {
+        if (user.Id.Equals(default(Guid)))
+        {
+            throw new BadRequestException("Invite the user first.");
+        }
+
+        var originalUser = await _organizationUserRepository.GetByIdAsync(user.Id);
+        if (user.Equals(originalUser))
+        {
+            throw new BadRequestException("Please make changes before saving.");
+        }
```

Comment:
Does this actually work? I assume it's comparing object references, which will always be false, but the intent is presumably to check value equality. (Note that the unit test uses the same object reference, which you'd never get in practice.)

If it doesn't work, I suggest just removing it as it's effectively dead code and I don't think this check is that important.

Reply:
It does not, I just tested it to be sure.